File: 035Menu/app/src/main/java/com/example/a035menu/MainActivity.java
package com.example.a035menu;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import com.example.a035menu.R;
import com.google.android.material.appbar.AppBarLayout;


public class MainActivity extends AppCompatActivity{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        AppBarLayout appBarLayout= findViewById(R.id.appbar);
        Toolbar toolbar= new Toolbar(this);
        toolbar= findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_mio,menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        int id =item.getItemId();
        if(id== R.id.menu_settings){
            Toast.makeText(this, "Ajustes", Toast.LENGTH_SHORT).show();
        }else if(id==R.id.menu_settings){
            Toast.makeText(this, "Desajustes", Toast.LENGTH_SHORT).show();
        }else{
            return super.onOptionsItemSelected(item);
        }
        return true;
    }






}

File: 044recyclerview-en-android/ListAdapter/src/main/java/com/develou/listadapter/MainActivity.kt
package com.develou.listadapter

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.develou.common.Article

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.articles_layout)

        val articleList: RecyclerView = findViewById(R.id.list)

        val articleAdapter = ArticleAdapter { article -> onItemClick(article) }
        articleList.adapter = articleAdapter

        articleAdapter.submitList(Article.data)
    }

    private fun onItemClick(article: Article) {
        Toast.makeText(this, article.title, Toast.LENGTH_SHORT).show()
    }
}